Orm/Xtensive.Orm/Orm/WellKnown.cs

```diff
@@ -95,6 +95,11 @@ public static partial class WellKnown
     /// </summary>
     public const int MaxNumberOfConditions = 256;
 
+    /// <summary>
+    /// Number of records in multi-record INSERT SQL statement
+    /// </summary>
+    public const int RecordsInInsertBatch = 10;
```

Comment:
Is 10 a good constant? Why not 42 😄 ? I think we need to make some benchmarking to choose a good default value or convert this constant a configuration.

Reply:
We can calulate optimimal value. For eaxmple if we have `.In()` with 259 items then:
`RecordsInInsertBatch = 10` case:    25 batches + 9 single queries = 34 INSERTs
`RecordsInInsertBatch = 20` case:    12 batches + 19 single queries = 31 INSERTs

not a big difference.

For better optimization we should precompiles batches with difgferent size: e.g.  8, 16, 32... and combine them.  But then we will have bigger compilation overhead.

I think it would be micro-optimization

Reply:
Have you seen this or something similar? https://www.red-gate.com/simple-talk/databases/sql-server/performance-sql-server/comparing-multiple-rows-insert-vs-single-row-insert-with-three-data-load-methods/

Thought it is quite old it raises question about sweet-spot number of values.

Also we have ```BatchSize``` setting for session (```SessionConfiguration.BatchSize```) and constant value ```SessionConfiguration.DefaultBatchSize = 25```

Number of parameters also very across RDBMSs, we did batch sizing that takes maximum number of parameters into account, it could be important here too

Reply:
Thank you for the link.

When we don't know exact number of elements, the best estimate for batch size would be SQRT(average # elements).
When our threshold for temporary tables if `256`  the relatively efficient batch size should be `>SQRT(256)==16` 

I'm going to make Threshold value configurable instead of hardcoding and to calculate batchsize as SQRT(Threshold)